lxd/container_lxc.go

```diff
@@ -1402,24 +1486,54 @@ func (c *containerLXC) expandConfig() error {
 func (c *containerLXC) expandDevices() error {
 	devices := types.Devices{}
 
+	var storagePoolsInProfile []string
 	// Apply all the profiles
 	for _, p := range c.profiles {
 		profileDevices, err := dbDevices(c.daemon.db, p, true)
 		if err != nil {
 			return err
 		}
 
+		// COMMENT(brauner): Check all pools specified in the
+		// container's profiles.
 		for k, v := range profileDevices {
+			if v["type"] == "disk" && v["path"] == "/" {
+				if v["pool"] != "" && !shared.StringInSlice(v["pool"], storagePoolsInProfile) {
+					storagePoolsInProfile = append(storagePoolsInProfile, v["pool"])
+				}
+			}
 			devices[k] = v
 		}
 	}
 
+	// COMMENT(brauner): Check if only one storage pool, the container is
+	// supposed to be created on is specified in its profiles.
+	if len(storagePoolsInProfile) > 1 {
+		return fmt.Errorf("Profiles the container belongs to specify different storage pools.")
+	}
```

Comment:
My impression was that when we have different profiles that each specify a root device but with  a different `pool` property set, then we need to ensure that all of these specify the same `pool`. Only in this case it is fine to proceed. Afaict, profiles weren't allowed to specify root devices before and to be honest, in my opinion, this is going to pretty fragile code but I certainly get the point of doing it via profiles.